bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java

```diff
@@ -144,45 +161,89 @@ protected GdbLaunch createGdbLaunch(ILaunchConfiguration configuration, String m
 	}
 
 	@Override
-	protected String getGDBVersion(ILaunchConfiguration config) throws CoreException {
+	protected String getGDBVersion(ILaunchConfiguration config) throws CoreException
+	{
 
 		String gdbClientCommand = Configuration.getGdbClientCommand(config, null);
-		String version = DebugUtils.getGDBVersion(config, gdbClientCommand);
-		if (Activator.getInstance().isDebugging()) {
+		String version = getGDBVersion(gdbClientCommand);
+		if (Activator.getInstance().isDebugging())
+		{
 			System.out.println("openocd.LaunchConfigurationDelegate.getGDBVersion " + version);
 		}
 		return version;
 	}
 
+	private String getGDBVersion(String gdbClientCommand) throws CoreException
+	{
+
+		String[] cmdArray = new String[2];
+		cmdArray[0] = gdbClientCommand;
+		cmdArray[1] = "--version";
+
+		List<String> cmdArgsList = Arrays.asList(cmdArray);
+
+		IStatus status;
+		try
+		{
+			Map<String, String> idfEnvMap = new IDFEnvironmentVariables().getEnvMap();
+			ProcessBuilderFactory processBuilderFactory = new ProcessBuilderFactory();
+			status = processBuilderFactory.runInBackground(cmdArgsList, Path.ROOT, idfEnvMap);
+		}
+		catch (IOException e)
+		{
+			throw new DebugException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, DebugException.REQUEST_FAILED,
+					"Error while launching command: " + StringUtils.join(cmdArray, " "), //$NON-NLS-1$
+					e.getCause()));
+		}
+
+		String gdbVersion = LaunchUtils.getGDBVersionFromText(status.getMessage());
+		if (gdbVersion == null || gdbVersion.isEmpty())
+		{
+			String errorMessage = status.getCode() == STATUS_DLL_NOT_FOUND ? Messages.DllNotFound_ExceptionMessage
+					: status.getMessage();
+			throw new DebugException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, DebugException.REQUEST_FAILED,
+					"Could not determine GDB version after sending: " + StringUtils.join(cmdArray, " ")
+							+ ", response: \n" + errorMessage + "\nERROR CODE:" + status.getCode(),
+					null));// $NON-NLS-1$ // $NON-NLS-2$
+		}
+
+		return gdbVersion;
+	}
+
```

Comment:
I see some important code from DebugUtil.getGDBVersion(...) is ignored,isn't a problem?

```
// Start a timeout job to make sure we don't get stuck waiting for
		// an answer from a gdb that is hanging
		// Bug 376203
		Job timeoutJob = new Job("GDB version timeout job") { //$NON-NLS-1$
			{
				setSystem(true);
			}

			@Override
			protected IStatus run(IProgressMonitor arg) {
				// Took too long. Kill the gdb process and
				// let things clean up.
				process.destroy();
				return Status.OK_STATUS;
			}
		};
		timeoutJob.schedule(10000);

		InputStream stream = null;
		StringBuilder cmdOutput = new StringBuilder(200);
		try {
			stream = process.getInputStream();
			Reader r = new InputStreamReader(stream);
			BufferedReader reader = new BufferedReader(r);

			String line;
			while ((line = reader.readLine()) != null) {
				cmdOutput.append(line);
				cmdOutput.append('\n');
			}
		} catch (IOException e) {
			throw new DebugException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, DebugException.REQUEST_FAILED,
					"Error reading GDB STDOUT after sending: " + StringUtils.join(cmdArray, " ") + ", response: "
							+ cmdOutput,
					e.getCause()));// $NON-NLS-1$
		} finally {
			// If we get here we are obviously not stuck so we can cancel the
			// timeout job.
			// Note that it may already have executed, but that is not a
			// problem.
			timeoutJob.cancel();

			// Cleanup to avoid leaking pipes
			// Close the stream we used, and then destroy the process
			// Bug 345164
			if (stream != null) {
				try {
					stream.close();
				} catch (IOException e) {
				}
			}
			process.destroy();
		}

		String gdbVersion = LaunchUtils.getGDBVersionFromText(cmdOutput.toString());
		if (gdbVersion == null || gdbVersion.isEmpty()) {
			throw new DebugException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, DebugException.REQUEST_FAILED,
					"Could not determine GDB version after sending: " + StringUtils.join(cmdArray, " ") + ", response: "
							+ cmdOutput,
					null));// $NON-NLS-1$
		}

		// Used to test if version comparison works
		// System.out.println(gdbVersion);
		// gdbVersi
```

Reply:
Hi @kolipakakondal. Yes, it can be a problem. Thanks for noticing it. In the new commit, I replaced getGDBVersion with getGDBVersion from the DebugUtils class with a little modification for DLL not found error handling